### PR TITLE
Fix #2856 - Fix rizin on Windows over OpenSSH

### LINKED ISSUE #2856: Console on Windows through OpenSSH incorrectly detected as "xterm", breaking Ctrl+C and others
When using rizin through SSH (https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse?tabs=gui) on a windows host with PowerShell as the shell, errors about `stty` not being recognized are printed and arrow keys don't work right. I have also observed some cases where Ctrl+C didn't break.

Bildschirmfoto 2022-07-31 um 10 26 56

This happens because `TERM` is set to `xterm-256color` in my case, so this condition is true:
https://github.com/rizinorg/rizin/blob/f1b2e7bd758f24dfab2577d615d3b86b4c31a60b/librz/cons/cons.c#L1549-L1553

Hardcoding that to false seems to fix all the described issues. So I guess some kind of better recognition logic is needed there.

### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Windows 10 19044.1826 x86_64
| `rizin -v` full output, **not truncated** (mandatory)  | f1b2e7bd758f24dfab2577d615d3b86b4c31a60b (dev)

### Steps to reproduce

* Set up the OpenSSH server as described on https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse?tabs=gui
* (not sure if necessary) set powershell as the default shell https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_server_configuration#configuring-the-default-shell-for-openssh-in-windows
* Connect through ssh from a client where `TERM=xterm-<something>`
* Run rizin and observe the behavior above

### PR DESCRIPTION
**Pull request by GustavoLCR**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fixes for rizin under terminal emulators such as `mintty` and over an OpenSSH connection.
Stops using `TERM` environment variable to determining VT compatibility.
Uses stdin file name to detect `msys` and `cygwin` ptys

**Test plan**

Tested manually

**Closing issues**

Fixes #2856

### CODE REVIEW

File: librz/cons/cons.c [hunk 1]
```diff
@@ -1457,7 +1458,7 @@ RZ_API int rz_cons_get_size(int *rows) {
 		I.columns = csbi.srWindow.Right - csbi.srWindow.Left + 1;
 		I.rows = csbi.srWindow.Bottom - csbi.srWindow.Top + 1;
 	} else {
-		if (I.term_xterm) {
+		if (I.term_pty) {
 			ret = __xterm_get_size();
```
Comment (XVilka) [inline librz/cons/cons.c]: If you rename `term_xterm` to `term_pty`, maybe it makes sense also to rename this function?

File: librz/cons/cons.c [hunk 2]
```diff
@@ -1536,6 +1537,20 @@ RZ_API int rz_cons_get_size(int *rows) {
...
+typedef DWORD(WINAPI *GetFileInformationByHandleEx_t)(
+    _In_  HANDLE hFile,
+    _In_  FILE_INFO_BY_HANDLE_CLASS FileInformationClass,
+    _Out_writes_bytes_(dwBufferSize) LPVOID lpFileInformation,
+    _In_  DWORD dwBufferSize
+);
+
+static GetFileInformationByHandleEx_t w32_GetFileInformationByHandleEx;
+
+static inline bool is_win_10_creators_or_above(DWORD major, DWORD minor, DWORD release) {
```
Comment (XVilka) [inline librz/cons/cons.c:1549]: Add a small comment, please why do we check for this, what has changed?

File: librz/cons/cons.c [hunk 3]
```diff
@@ -1575,8 +1606,16 @@ RZ_API RzVirtTermMode rz_cons_detect_vt_mode(void) {
 		if (info->release) {
 			release = atoi(info->release);
 		}
-		if (major > 10 || (major == 10 && minor > 0) || (major == 10 && minor == 0 && release >= 1703)) {
+		if (ENABLE_VIRTUAL_TERMINAL_PROCESSING && is_win_10_creators_or_above(major, minor, release)) {
```
Comment (XVilka) [inline librz/cons/cons.c:1609]: Or here

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by XVilka** [comment]
@GustavoLCR it also seem to break Windows unit tests:
```
Summary of Failures:
 21/111 rizin:unit / compare                               TIMEOUT        300.21s   exit status 1
 22/111 rizin:unit / cons                                  TIMEOUT        300.18s   exit status 1
Ok:                 109 
Expected Fail:      0   
Fail:               0   
Unexpected Pass:    0   
Skipped:            0   
Timeout:            2 
```

**Review by thestr4ng3r — APPROVED** [review]
Tested and works on my setup. Thank you very much!

**Comment by thestr4ng3r** [comment]
@GustavoLCR The history looks nice, should we rebase+merge instead of squash here?

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
Let's keep history for future generations.

**Pull request merged**